default/python/AI/turn_state/_empire_resources.py

```diff
@@ -100,13 +109,25 @@ def set_have_nest():
 
 
 def have_computronium() -> bool:
+    """Do we have a planet with a computronium moon, which is set to research focus?"""
     return _get_planet_catalog().have_computronium
 
 
+def computronium_candidates() -> List[PlanetId]:
+    """Returns list of own planets that have a computronium moon and a species capable of research."""
```

Comment:
https://peps.python.org/pep-0257/

> The docstring is a phrase ending in a period. It prescribes the function or method’s effect as a command (“Do this”, “Return that”), not as a description; e.g. don’t write “Returns the pathname …”.

```suggestion
    """Return list of own planets that have a computronium moon and a species capable of research."""
```